fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/predicates/Value.java

```diff
@@ -54,14 +57,34 @@
  * A scalar value type.
  */
 @API(API.Status.EXPERIMENTAL)
-public interface Value extends Correlated<Value>, TreeLike<Value>, PlanHashable, KeyExpressionVisitor.Result {
+public interface Value extends Correlated<Value>, TreeLike<Value>, PlanHashable, KeyExpressionVisitor.Result, Typed {
 
     @Nonnull
     @Override
     default Value getThis() {
         return this;
     }
 
+    /**
+     * Returns the {@link Type} of the scalar value output.
+     * @return The {@link Type} of the scalar value output.
+     */
+    @Nonnull
+    @Override
+    default Type getResultType() {
+        return Type.primitiveType(Type.TypeCode.UNKNOWN);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Nonnull
+    @Override
+    default String explain(@Nonnull final Formatter formatter) {
```

Comment:
Maybe that one should be `describe()` which is more in line what other Das do.

Reply:
Alright, I renamed it accordingly.